server/apis/v1/promql_service_test.go

```diff
@@ -337,6 +337,72 @@ func Test_PromQueryBuilder(t *testing.T) {
 			}
 		})
 	}
+
+	// tests for pipeline gauge metrics
+	var pl_gauge_service = &PromQlService{
+		PlaceHolders: map[string]map[string][]string{
+			"vertex_pending_messages": {
+				"vertex": {"$dimension", "$metric_name", "$filters"},
+			},
+		},
+		Expression: map[string]map[string]string{
+			"vertex_pending_messages": {
+				"vertex": "$metric_name{$filters}",
```

Comment:
I'll make test rules rigid in e2e [PR](https://github.com/numaproj/numaflow/pull/2277). Meanwhile, can you update the expression here to `sum($metric_name{$filters}) by ($dimension, period)`